app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    update(posts_counter: comments.count)
+  end
```

Comment:
- [ ] Kindly correct this method, you are to update the post counter and not the comment counter. 🙏 

- [optional] I think you can make this method to be private because we only need to call this method from inside this class where it's defined (  **You can implement the same for `comment and like` models as well.**).
- [optional] Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the counter method every time a new data has been saved to database. ℹ️  **You can implement the same for `like and comment` model as well.**